### ISSUE TITLE
The effective HP calculations are incorrect

### ISSUE THREAD

**Issue by TornOne** [PathOfBuildingCommunity/PathOfBuilding#5178]
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### What is the expected behaviour/value?

When the damage dealt per attack is ~10, and I equip a helmet that increases my max life by >200, I expect my effective HP to increase.

### What is the actual behaviour/value?

The effective HP does not change.

### How to reproduce the issue

1. Modify the Modules/Data.lua file, line 478 to a higher value (I'm using 1048576)
2. Import the character code
3. Go to Items > Helmet dropdown
4. Hover over "New Item, Ezomyte Burgonet"
5. Observe it increases total life, but not EHP

### Character build code

```shell
eNrtPWtT48ayn-NfoaLq3NoUNmhGby6cU8aYR8IrwC67-bIl7DHWIkteSYZlU_z30z2jpy3JsmG3krr3pE5ipO6e7p5-zoyk3f98m7jSIwtCx_f2NsiWvCExb-APHe9-b-P9zWHH3PjPv1u7l3Y0vhjtzxwX7_y79csu_y257JG5exsGoEV2cM-iDwkp5TNcm9peNGa-d2Z_8YMjf7i3ce57bEO6s72hE-1tdF0nsDekgWuH4bk9YXsbt040GG9Idjhg3rCXXe-7bMK8yHadMNqQJrbjXfuDBxYdBf5sCoxvSI8OezrzhwB7cnZ5cXWTG93x8qMD97_sXrr2MwuuIzuSQvgXsAJKsO_ZsQPkH213BnQU2SLU2DItU1FM3djYLse8njI2TJG0LUu3qCIb_N9mFdJlwPqjERtEziPrBU7UG9vegKVU9Cq8VWHPZm7kTF2HBSk8qYK_8UG_B5fXGaSpalTTtgydKla1LALPzxQnV0HC7I73XVDXGqMg7sm950RsTeRL3wl97xXy5VErRezNXBdcpBmsP7lzvLW00Q2YfTESJnRlD51ZmFFQq5DObM_u-WE2UUalUZ86I1YArRSif90M7gq8qxkksnnJAogAUTMEZHYlhHiEazbwIcqsMsYqKDk51htsDcz-9ToIawx0HeUiCrUqZ519zUOaShXgAfuWRTS1hlweUKu03xMvk4GqtIZeHpKopJq_Rz-CzLZcMzxc9I8vM6o60dQtzTIsalimXjfpOZ4NvSbhDHYQ-sQbZIMQWkf4vRewkAWPuUxVNwSiXIE1YFK8c1ljnGyY2Koy_mS5frB75sUjPmfmostbmm7WIZ4yNhgfQT1wZUesmU_mpLHkWiUjdEHJilZHuFTJltwMZQWFIWK5wjqEWFtmHd6K-rq1g2GDIOKx4P75euwwd0XoRIKePW0Qc3BO8tj5uaG02XhlamvE6oqa6z_aYT5eENXS6gUTCHmZqi2AQRUFCEM2VwwSc8vQIB5TohuaTqrLT_8LFp_ua4h0g4k_y2K7JtdHLAFemLJK-S7Hz6EzgKrLnkDZcMWGs0Eh-JItKMpNVbcUwzK0ynSx70KTMC9c3Sy4bhkKrS7DosgePBz4w_t5Hcq1o6yOcT2bTiFUoIU0xTt0AlBd6OQLPm058AW4Rt4dq2Nez0eHajhABrwwQE2Avx9HHpTTjUeZw5gfilYaMyjVD-eG6Si0Afj8GNURIWn4ziD6YDfLO0foWXNmTSormkNoLBp1FRywYbdy6T8B82Ns3cPVoM_srAyrLOoOA-Z9f25MvwDeaIC-N5wF6A2Nx5jHKBvmxplAfA3DAzuypZDZwWB8ChO9t7GR--vQdt07cH64ili723xFBH-dTKZ-EPGLPdsdhJzmiTedRZLHFzMmTjj4fDcbjXC9AmhGAV9u6R8e9ns3Jx_6MRt5lPDBcd3P3mxyh528-G_Wyl8zHh6lcHYXip97Gx8c9nSNWAcssh0XVDPwXdeehmy4tzGy3RBGduAnh7kGeQdRDTUOdexEIiBX0coAqin1v7EgAnmhshgEDqvkK72_hCkxIOZznLYqarg8UU1IhPGeHUYit1doii_0VFPB9ZZKcfBmDS7YjO1WjhzfXaKJ6HnKQsi0zsgZoMXXT_kNQAuoGr0MBuAsg-ea-Y6TUTUNvt5TRUDcrEYWazZV2PHdGq3y9aJKrYq71egHbGBXyi5uViOn5aLv8UXFciopVA2lc9_jRg5O03VcTByVM9t3WQpSTfAiGrNAZKNKSmcQoxKQWscJnLtZVO3GOYgaXfHWt0JDeK8aVTR0FTLgvZpIVOhaKhSah6kmJdqkykBWh5pU0RXqE3drhEjajAr-49vVBEQRUzl_cUlU599QMFf7N96sRhb-sUgiCmYNKXBbPmAjBlGo1phTmJooGs28Awh4UY1CG5ISWan76DtDUay_ihoXspxYpqmVaIkgUaq3lSnyZua1qp_rcd5E_Tf2A6uklAdZQggC73FNam9GKa3-j5nt4m6E776OIFSuzmQ2Ac7eQk4_Cm1veIDLI40F3d1O6ttdXlKEUgiF7xGbhPvP0H4cYtCc2-WY2BHkOzY5xY3DGx-LcHsQseBUbCQK27O5nuLCNOK7ewXCCdyQjeyZi9f_mNmuEz1j3Z-7GhOlcDEc-09oXoIM1j4hJNaLq7PuqbjZdaOYCA6TiC30FTPCdUD4tfgi34jsZtxydXB-HW_gzobsxIvbtlSPrn2HPOE2K66spv42Rykd6Jdd4CcGPnL9O9ulCYqbiZe_T5L7vGc4wXlzQseL2H3Ai9uvQkx-Q6hqQ7pnE_z7DFqFITQ92ycRKGEbNbHN2YFfPZioB9zaOhUx6GtO5XOyDPyZJ6YNmxdc_MH6N0aXYnzO3ZnjCfefTSY-2MglVJORHU6Epb6J8PGs93wPl1dRBUPhigUWUODE1soUxEEv7UBA1KkrNbKyEVdVWo6EVMI1L-UjcJ87P8L1lbdXG98_HtjhD9dWbqAVlcRTh9QfjP2frJtjCJjIsVg39P0fr6SyEVfU1gkEAi90Rs8_WVnd4ZAN09W5ZBnhx6qrfMwVFcaJSCkVqYT1Cu3titj5RvkihD5gwPY2bgLGdlSLGK9LIQn5ptN4GTgT2-2yeycsn58V1SroSYLgj9DX6_NrbobnRl_B-DGvpcZz5Ltsski40uYbpOXyEVYNoZxIzsgXGc1y9NxY2z9NjVfQCf4wDeaIr6e86wiUU6s4DvGzlYb7Oj9MaYcuaOFVWuMUarWWG-PNk5IYIz1bKNY6bNERr6qYNOtUEl1ZRykNqRtMnv-mUbKxzrvBwPZYz_Xth7XNroxG40TOcSWB_Pb1Dad-PQvu2Stsp4zKivIJ5DeXD7oPKM9xK2Bt4RZINPaDBPHNpTrxBgGzQzbMlv3WFq-aVvPaPKYgpST-4T5_C2KM1_b2InZTLQqst--E4f-2Cx35ozN4paX0oVXwbHcf95ihjlrdHwS-lBL4h1sJT_EHdjh-becX280ivaaaFeWIQP3JC1TQZzpoV2hj0EP6T96b9cHiXO-t4w1XL9M4qsRx_-FWBq7Lj5E9gvjrr4KWESkqMxaxYj0vw15QZ_zzmkV8bV0sqeNPbPmTRXFOh8QL0_BH_lEZZJ6iEkMIEM_x7u3exl-qaqlmW9NMU39p_6UpxIS_LFlX4S-imDptU0umeMs0id6mssJvUVM35bYmAyQCUkK1tmroKkJSVW7DL5XAb1WlltkmhgV3QFp_iGvrKjHVtqaqqtHWZdWgbaIR3WhTRaZGm8iEAC2AUdqabsCYqqkgGwaM3dYIkc22CsORthiVmrKht4kqU9KmxNJgNEuT4S5RqQwwmqK3daIaeEVXkTJyBELpcEVWSNskitbWDByDKAACyrAsGEmlhtXGxRRAp3CZUhl0oxiabrWF-CqMBsyoGrVgUEOhbaFNVUFpdFOHkakCIsG_rLZi6ahpVdfgikkMtS30a6mIaFFNbau6CsSFphUYlXA9wHWCWqI6NWS4DpyBZlTgX1FU4J0QQkFq1QJJNQM1CYrE-VANCxjTZJTJQBE02aQwhyrKqssaDKtYwKkhK4CvyDIwqVsqcCDMQFOAMyKjkgi1iIKzSoAD0AHonVpw0zCBSR3wARx-gn6IacBNMAFKFAUnAGZNgSGBhqUjt7pMQbuGqgAJPvmGAZcNS0ESloXqk2UYRQXGgZAJIDCeBhcIaE4DydFsFAAE6wCTjsADco-WESt-aAwtXol3ZN5fnYpYMY6iabizvf309LQ1hVrAH7Fvjsu2Bv5kewpI4EYdHhU6SHa7C__bvz_rffq2_8k4uDcuiff9N_UidC169ngx_fqHyR7867PHj7c0PA8_fPj28BSd_vb01Lnsjk4_0qjnHR3udwejb53u9Rn9bs-Gfxz3v398fH_458lDePjUOzuSg9Hh-eHvnff9PyzfHI-HZ4czWz2_3b96VOlvzs2npyObjNnH7perSTj5_aT751S_eIyez--nYcAOpp8U2aJ92_h8EX0ZhWf27x-f_Y_T_X5k_6k-y3_uf_hu3nz6rTd4PPOOddm_vOtdXh167PFBpVPz_ung6T056YeK_nhzenZ7R0YnV-8Ho-7sTzr97Bw9T_r-8aeLw7uhO9I7qn3R7Z7Mxv7jx_efboM_giPqdU_og_7odqPbc-fPyy8XH7ta8OXy22XwrXvZ3RNRLNH-rnguL9yOIxqEKB7NMIbhj3M_YvxEHF5M_uARDg-sxefrrqMAk8N335982tswLWPLMtDXVM3QxXbdMbOjM3uaBHmEjHf7rHizL7KjAwdCYMD3ivOAH_nhemvLBEc2dEMl4hjdLg_3cZzNr0bPQhZnUGZPfY9fzm0PIijfGtSEIV7ZAYT3Hemqe9VvXfuuBPnQbZ1M7mZQ2N_a3rDVC-xRxIY7EvLUugzYyPm2I_0V2N4925G3wKVf5latBcyFJzgwSnCo_nLqD2yXL1cXj0_TFByff2xdz0ZFXNN8STuPnthdYENtEQ7Cxsv-LBCHkZiSAnCq8ebpjkTlVmwCO9J-B_5p8Ym5Yl93JM0CPUxdZ-DgXdL6K7Lvw50Bz1efh5xsO_6PuPgSD05e3ilKRzF-_RfWFHGTJDZ-BDctkDuUVCnyJVNKxE_uWTSPNr-U36Jq_jZqQOIqaClW_gaKnsP6ayBm8SXVj_bCmXind_RfkZF3ht4x4Of8gHiP8x5yhznzh1dIQeJ0RN00TM92Lt6HgQQEXQqhLIVQl0JoSyGSY_-8Xiq6BCl1iSDwnyQw5UcWtA4Y1NX_A1Yd_m8oHfuBt8w7tBdR4KVa5arEIo3bGSkae73lG4b2wk9yi4m5iA-RZg_Nks-LSJZOcm5QBmEYZR5lNPMRXV_TR2Bc3QQ_IR1Fq_aTTWL8Cw2Qi50YJJ6elVByKRO9pZAKt9gkSIBzLPkjCaJbzsTFYKIvwElpKXKdE3HhWLLOOSdfer3tJvRfFt3OIhpIrXVU9ddaNy-3YdDZP8bfQNKlHqeUeJw4RCgdM3fCola6ejAL7sErovj2jgS1Xfx7HzQYP48GhdOOJC93Sp58UqOveH7nysaT38or8WPOtDIyKQU8Nmosuqb2ArUF8-6jsVZ2M30UkYn1NzwOrswDkpe5p0zUcs--6vB_8r5dyH9ya1OReTKIWWptavxvMQkto5CX4oubhoYgE3H4S0KGW0QvGv6ISXkhJDw_0tqkFnf8zC-ynYl6i_zHZ6JHcAYbm_EYzCz6yPvzkz_e91t8YwAD2gFYFdfKqT34OoMCcigd2cFdKz5LC0FZgWmO_5p3lY68RTWtJc4NS-LgMJiFQVr5o8RlWGbrg-BzRwLDlsiWsiUXL8GFxUva4iVi5a_1ZgGYV9QSz3iAMOkdbV7CUjM-6qT_5Es5WjTlv2Il70CH3lbyRdE7aOc70KEWonOsPQnKYamgq4wQNLc5MhamNlnuqPLKhICjPD-b7yzgh8i_5v0Ij4pnGEob-ugMh8oGYEHPwMWoRpsb6J0qd-bERviib2KEyVF4uR0zTxo7UVsikDjRGkWKdkIpwoOX0ijwxbjSHRv5MN88BKQUVFD-ephCaMJRi3wC41MoEELeBLXOHFA1lkoAhYde1g8elvkG4QMm580DSA2EsRTCXAphVadvvSR9n0NfjACtI9917GgMvjrzIhecM03dKqUVqZssT92FnKnOQxCRmpGuGKHrcUDyNllcXcyuuWcrS3Jv9qCqWpaYi_m7cWbWlPnMTHOZWKdlmVg3FzJx5jISVUUmznlOkoJRhHz23aRyrib_v5eW9cQtytLyle14d_4TPiQ5ZK2e732ZBaDSfd8H-59Ls6pSm2Xl-Qy7kE7JFpm_RLf0xQyrrp909XkZqrvC4tqJUpdwC7ny5Z3ewaUS8WweWhZ_iqbYCGZ5tpizOnpjTD23PEO0DpUbYy6kZBNyq7w0Jbf1DMuUEQ9SrL5KSoYEDnzOlRDlZcNcEYDIMBoxGyGTdsJw4tzYJmcHenJHjTIk-rJpNoUVs66_0EJ7nTxZHnfqhVqKajWQPyrSkDeIMzDTf8v8XacxeTkIWRIVacJwWVTsRt-dwEnWzQ59586F3G7PbKk7mUFpUBvDqoMTLdDIgg8pXZSagKulDvLyjsodpVm5m8POKt-4-S36cQkkeAmEikagpOC-ZofQAnsNhsqkI3qHqquhk_luQF9bPbkoplP6ksTLBtTiXQ2o2XE5Thrbj2APEAoCLMewmkke0AXcrzOoTPjTw-W2-_coPN4gGIAOVwwHlW6qxkpRSv0UF1Vxp3LaOvN9L-RnhK-g-m1QYxBj_SJDnR8uc2VKy1z5C3uCPM2C589DfLoyZJnlb_LcrhWS7FzW4yRs_hKFdoj5pJg2sW8umKp430KSpHIL3Al2uQdpHfPXijXpOiLz6xEauM46dHIllkqNMrkWCM3HA5JzvFW8H5t6rXHoCJh4WKTtQgNSyr9OsEbk4TBbKOAPdkr4cqzZFIaww3iBoEizyJmaKwEbEMyFyhKLK511XHhpYn419OZqzWbmXENPKdKDwK4up5ds1mRC3jB8UQwUeFAzOx6G5lDC3ZLEQMQBHgnm9pM_q6BCS6lo1VRKZXjXQVsuOoXYwomxk_OorTJj4mtzUFWLBqA5BTDq957NbwP03bM0J8rfOQdhBPgpRalOllSl5AfUpHUgy_VKliuWLF9WIFqDlUOyXLfpG7WW1NmElCbwAzDmiXQY2PeiLrq2p9MxX79ZlsXrquwilVxyVuuTc5C2glnc3UwLb1xC8t1hfgUpl02KnWJV8arJjdNMOVslQVc2FOBSwcX7ci57tuf5kXTHQNH-d-blW-LsXm8M8aSiV_05hyZAkJ9zbIKWLAPvMxdbMhvf3yHxP165sluAwJnWlxyXyDZP9QZ7orUrt5nJq_Mmn0-tVEObWVhm3VS04r6purgpuqnShR6xegH2__dGlTTslvYxYybdOiGTLuyH1r4TTmbRGB_eCx8abFESrVlcVOYo51dGM4Mxq9dBX6DKwNnj6HHlITbSkmooOYHFK1OoVJk9GCdLbdLID6SnsQNXnvG8xHtvAGEaS6iSpTjcVnP9JxZG-ZqqdHz9Jw1PyoePx7yEGM7fJBFKmAlypF7Lxxi8psBIod5uyFPaTf1IxpR1GBNd5w9g6x9_uG9us_Kse3TSa-GjHp07f-hgBwAiR87ISSoJoWvQXxcS-aMjGoKlWYxjpYkqeXlzd4IPWJRtECI4f7Fgzw-C2RQ3G_dd3x-eTCYzDzilFYFFk4uRRSscqxHjSfHweDpUyfw6YUoctTkCVqJQSsYTmyHAD5obGgsh8Y5gKDkjPF48RPtxWQpVRiEWBs_QoTD1hPJd1RziG-WzUoMwygyi_82eOFA_JovV146LhwdSW4B6RjqeTQDm_s4JhqvaQ9JYqmX1irAFuJ6ewrxF_eAruGC4KkNQ545pUTNvCGknS5TytR9pmI8sjE_FGzl7qdbNJlpHi3IiVlT7jR9EPmT1N9M5yTQudskbqZs0UnexM4nPNP5kVVtNVP2bPZzTc9eLmOtPf4ie4wNQP0zRyQErcWrjJyuc1h2H6X_3J89g0gsHWQmV1zzJSlY5wWK-CjvmylgkUujarOJjFtV_ND2CaswddKFy_swpkeWyoy6WtdBf_ay2KGcW166fvA9bPAgjYdsfiac2k1fC54CEkSoZTLoWlAM6ciFzhxmMvgiCzXaOCK0aScsBWXkg8UxR8gxSHiUnAKmUQM0BmVVANAdUIkOssDzUIhCuRhVASmD2_eGzlLy0NgE0K2jlxVPKaPlRTvXWIoTYic5AkgmMn8-qeTar-LbGRcu5frKnUvfuOQyhQBcOk5_AdLG0BHUejTZDW7DVRV2syVCZLEuYQpteEaXSnUoNrYwnZT39NtMCfSM0dT0m9fVGa4hWpk2yZIbLDGrJDIhHRFYdqMSU9HXGWWJ_FZG6AWFlfTci6wiirmcLZD3LU5dPz2L2KtHsqnTL7GtNyRtFnpL0upyhNcKgskzuNAGuOIHlal7LwrR1kBo5ZRXMSjVMA7G1tbVM17BLfT3XUtYzZ2V1o1PXzwNrmDhZW_lKE08tKVibVaxLGX9FQdRwLhcK5MbzoK9hl-tPxCsMpmmQRtyGftMcdA3NVbUlDaIMXVu_a0aM5hO60D01EEdpoKiSnqvxRKhrq0tbJ_U18ZiF9rCxmS9hqaq13I57S_7aEn56ln_Jy_dGzv3Cd7mYxybPxXWe7PNcSsmHvNJn2vGVnv1vUz-cBQy_R5B9ByzdhS5Dx_EWXmmdDEjLvhxW8oLgPn9XVfx6vTsfLtlenNkXCcx_xD1jlH9OfRFBfN09AxNfea-QBbfim4hx4ITIc__4Er8Zc5F8M2YZ87g0wF-rdj1Goxj2ccwrhitv7vNydM5i_IH39KNrcvpl1gXQ_OfZE_hq4OzNE4ufdMtDu8lTbcXvxXU_9K-6R_06qfnnpfgrBpdJOklf_IA7dSCyF2VqWiJ6vKEZfx6kYMvnF-dlDDrJW-jj19FHuUHUSn357rCJqaQG63jNDNbmEQ1fWzdz-RL4ynruF9wycesVzJMTiG20oVmehPt-GJYKd-naAzYGdbGgxpCN_BeGy-DRmhdiWgOceQ_Q6lGqwqcefx92dzsJvrvblxCyLkb8U4sg9L9b_wX_xKBe
```

**Comment by TornOne** [comment]
As additional information, I believe the problem is with the EHP calculations speedup code. Setting ehpCalcMaxDepth to 1 seems to fix this issue. (Although I assume EHP calculations, such as when calculating the EHP values for the entire tree, would then be unbearably slow.)

**Comment by Regisle** [comment]
Yeah the speedup function is incorrect sometimes, I think there was another issue about it at some point, but we have no idea why its failing

and as you said, without the speedup, it becomes slow, I have thought about making it so the user can choose the values of the speedup func but decided against it as it would confuse too many people

**Comment by QuickStick123** [comment]
I just added a commit to https://github.com/PathOfBuildingCommunity/PathOfBuilding/pull/4915 that fixes this the problem was maxDepth wasn't big enough so it would hit a wall and not finish the calculation. I changed it to simply use the max iterations (number of times it does damage calculations) and maximum taken damage to decided when to stop / state infinite eHP. Current set at 100 million for infinite and 40 max iterations your example was about 32. Missing iterations will simply reduce resolution of lower significance parts of the number ( lower multipliers).
